packages/web/src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+
+import styles from './Icon.module.css'
+
+type IconProps = {
+  /** Decorator element to be applied to the upper-right corner of the icon */
+  decorator?: JSX.Element
+  icon: React.FC<React.SVGProps<SVGSVGElement>>
+} & React.SVGProps<SVGSVGElement>
+
+/** Renders a stems Icon component with optional decorator
+ * Ex: `<Icon icon={IconKebabHorizontal} decorator={<NotificationDot variant='small' />} />`
+ */
+export const Icon = ({
+  decorator,
+  icon: IconComponent,
+  ...iconProps
+}: IconProps) => {
+  return decorator ? (
+    <div className={styles.iconContainer}>
```

Comment:
spans maybe?

Reply:
Good callout I'll try it and see if it affects the positioning behavior.